psi4/src/psi4/libsapt_solver/usapt0.cc

```diff
@@ -0,0 +1,2459 @@
+}
+
+void USAPT0::initialize(SharedWavefunction mA, SharedWavefunction mB) {
+
+    type_  = "USAPT0";
+
+//    freq_points_ = options.get_int("FREQ_POINTS");
+//    freq_scale_  = options.get_double("FREQ_SCALE");
+//    freq_max_k_  = options.get_int("FREQ_MAX_K");
+
+    Cocca_A_     = mA->Ca_subset("AO","OCC");
+    Coccb_A_     = mA->Cb_subset("AO","OCC");
+    Cvira_A_     = mA->Ca_subset("AO","VIR");
+    Cvirb_A_     = mA->Cb_subset("AO","VIR");
+    eps_occa_A_  = mA->epsilon_a_subset("AO","OCC");
+    eps_occb_A_  = mA->epsilon_b_subset("AO","OCC");
+    eps_vira_A_  = mA->epsilon_a_subset("AO","VIR");
+    eps_virb_A_  = mA->epsilon_b_subset("AO","VIR");
+
+    Cfocca_A_    = mA->Ca_subset("AO","FROZEN_OCC");
+    Cfoccb_A_    = mA->Cb_subset("AO","FROZEN_OCC");
+    Caocca_A_    = mA->Ca_subset("AO","ACTIVE_OCC");
+    Caoccb_A_    = mA->Cb_subset("AO","ACTIVE_OCC");
+    Cavira_A_    = mA->Ca_subset("AO","ACTIVE_VIR");
+    Cavirb_A_    = mA->Cb_subset("AO","ACTIVE_VIR");
+    Cfvira_A_    = mA->Ca_subset("AO","FROZEN_VIR");
+    Cfvirb_A_    = mA->Cb_subset("AO","FROZEN_VIR");
+
+    eps_focca_A_ = mA->epsilon_a_subset("AO","FROZEN_OCC");
+    eps_foccb_A_ = mA->epsilon_b_subset("AO","FROZEN_OCC");
+    eps_aocca_A_ = mA->epsilon_a_subset("AO","ACTIVE_OCC");
+    eps_aoccb_A_ = mA->epsilon_b_subset("AO","ACTIVE_OCC");
+    eps_avira_A_ = mA->epsilon_a_subset("AO","ACTIVE_VIR");
+    eps_avirb_A_ = mA->epsilon_b_subset("AO","ACTIVE_VIR");
+    eps_fvira_A_ = mA->epsilon_a_subset("AO","FROZEN_VIR");
+    eps_fvirb_A_ = mA->epsilon_b_subset("AO","FROZEN_VIR");
+
+    Cocca_B_     = mB->Ca_subset("AO","OCC");
+    Coccb_B_     = mB->Cb_subset("AO","OCC");
+    Cvira_B_     = mB->Ca_subset("AO","VIR");
+    Cvirb_B_     = mB->Cb_subset("AO","VIR");
+    eps_occa_B_  = mB->epsilon_a_subset("AO","OCC");
+    eps_occb_B_  = mB->epsilon_b_subset("AO","OCC");
+    eps_vira_B_  = mB->epsilon_a_subset("AO","VIR");
+    eps_virb_B_  = mB->epsilon_b_subset("AO","VIR");
+
+    Cfocca_B_    = mB->Ca_subset("AO","FROZEN_OCC");
+    Cfoccb_B_    = mB->Cb_subset("AO","FROZEN_OCC");
+    Caocca_B_    = mB->Ca_subset("AO","ACTIVE_OCC");
+    Caoccb_B_    = mB->Cb_subset("AO","ACTIVE_OCC");
+    Cavira_B_    = mB->Ca_subset("AO","ACTIVE_VIR");
+    Cavirb_B_    = mB->Cb_subset("AO","ACTIVE_VIR");
+    Cfvira_B_    = mB->Ca_subset("AO","FROZEN_VIR");
+    Cfvirb_B_    = mB->Cb_subset("AO","FROZEN_VIR");
+
+    eps_focca_B_ = mB->epsilon_a_subset("AO","FROZEN_OCC");
+    eps_foccb_B_ = mB->epsilon_b_subset("AO","FROZEN_OCC");
+    eps_aocca_B_ = mB->epsilon_a_subset("AO","ACTIVE_OCC");
+    eps_aoccb_B_ = mB->epsilon_b_subset("AO","ACTIVE_OCC");
+    eps_avira_B_ = mB->epsilon_a_subset("AO","ACTIVE_VIR");
+    eps_avirb_B_ = mB->epsilon_b_subset("AO","ACTIVE_VIR");
+    eps_fvira_B_ = mB->epsilon_a_subset("AO","FROZEN_VIR");
+    eps_fvirb_B_ = mB->epsilon_b_subset("AO","FROZEN_VIR");
+
+    mp2fit_ = BasisSet::pyconstruct_auxiliary(dimer_,
```

Comment:
Instead of the above, the Wavefunction class now stores BasisSets that can be accessed with a simple `get_bassiset`. We can go ahead and get this in now and I can rebase this through with the correct changes.